java/java-impl-inspections/src/com/intellij/codeInspection/CastCanBeReplacedWithVariableInspection.java

```diff
@@ -182,6 +224,16 @@ private static PsiLocalVariable getVariable(@NotNull PsiExpression occurrence) {
     return null;
   }
 
+  @Nullable
+  private static PsiLocalVariable getVariable(@NotNull PsiAssignmentExpression occurrence) {
+    if (occurrence.getRExpression() instanceof PsiReferenceExpression referenceExpression &&
```

Comment:
The expression on the right side of the assignment could be wrapped into parentheses. Please, use `PsiUtil.skipParenthesizedExprDown`. Otherwise, a false negative will be get for this code:

```
void method() {
  String anActualText = "Hello World!";
  Object aText;
  aText = (anActualText);
  System.out.println(((String) aText).trim());
}
```

Reply:
Fixed